### PR TITLE
fix: consistent to_numpy behaviour for tz-aware

### LINKED ISSUE #4: to_numpy: make sure to not return object dtype for nullable dtypes for pandas pre 2.2.0

### PR DESCRIPTION
**Pull request by MarcoGorelli**
closes #4 

## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue # 
- Closes #

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

### CODE REVIEW

File: narwhals/_pandas_like/dataframe.py [hunk 1]
```diff
@@ -698,21 +698,38 @@ def to_numpy(self, dtype: Any = None, copy: bool | None = None) -> Any:
             # pandas default differs from Polars, but cuDF default is True
             copy = self._implementation is Implementation.CUDF
 
+        to_convert = set()
+        for key, val in self.schema.items():
+            if val == self._dtypes.Datetime and val.time_zone is not None:  # type: ignore[attr-defined]
+                to_convert.add(key)
```
Comment (FBruzzesi) [inline narwhals/_pandas_like/dataframe.py:704]: I am a big fan of comprehensions, I guess it's just a matter of taste, but if you want:

```suggestion
        to_convert = {
            key for key, val in self.schema.items()
            if val == self._dtypes.Datetime and val.time_zone is not None  # type: ignore[attr-defined]
        } 
```

File: narwhals/_pandas_like/dataframe.py [hunk 2]
```diff
@@ -698,21 +698,38 @@ def to_numpy(self, dtype: Any = None, copy: bool | None = None) -> Any:
...
+        if to_convert:
+            df = self.with_columns(
+                *[
+                    self.__narwhals_namespace__()
+                    .col(key)
+                    .dt.convert_time_zone("UTC")
+                    .dt.replace_time_zone(None)
+                    for key in to_convert
+                ]
+            )._native_frame
```
Comment (FBruzzesi) [inline narwhals/_pandas_like/dataframe.py:714]: Is the loop even needed? `col` should handle lists directly:

```suggestion
            df = self.with_columns(
                    self.__narwhals_namespace__()
                    .col(list(to_convert))
                    .dt.convert_time_zone("UTC")
                    .dt.replace_time_zone(None)
            )._native_frame
```

**Review by FBruzzesi — COMMENTED** [review]
Left a couple of non-blocking comments about style 😇

@MarcoGorelli and time-zones right now:

LoveYouAndHateYouAtTheSameTimeFabolousGIF

**Comment by MarcoGorelli** [comment]
😄 awesome, thanks for your review!

**Pull request merged**